sections/breakoutRooms.tex

```diff
@@ -1,23 +1,203 @@
+break-out místností též využijme existující Matrix primitivum -- state události
+\cite{GitHub-MSC3985}.
 
-Lorem
+\begin{figure}[H]
+    \begin{minted}[tabsize=4,fontsize=\footnotesize]{json}
+{
+	"type": "m.breakout",
+	"state_key": "",
+	"content": {
+		"m.breakout": {
+			"!roomId1": {
+				"via": ["example.org", "other.example.org"],
+				"users": ["userId1", "userId2"]
+			},
+			"!roomId2": {
+				"via": ["example.org", "other.example.org"]
+			}
+		}
+	}
+}
+	\end{minted}
+\end{figure}
+
+Pokud chce klient vytvořit např. 2 break-out místnosti, vytvoří dané Matrix
+místnosti a následně pošle state událost typu \mintinline{text}{m.breakout},
+jejíž příklad můžeme vidět výše. V poli \mintinline{text}{content} můžeme najít
+pole \mintinline{text}{m.breakout}, ve kterém klíče jsou ID Matrix break-out
+místností (\mintinline{text}{!roomId1} a \mintinline{text}{!roomId2}) a hodnoty
+obsahující informace o těchto místnostech. V informacích o místnosti najdeme
+pole \mintinline{text}{via}, což je výčet serverů, které je možno využít k
+připojení se k dané místnosti (ID místnosti nestačí pro připojení se k místnosti
+\cite{MatrixORG-Spec}). V těchto informacích můžeme též najít pole
+\mintinline{text}{users}, což je výčet uživatelů, kterým je doporučeno, se do
+dané místnosti připojit~\cite{GitHub-MSC3985}.
+
+Dojde-li k přijetí této state událostí jiným klientem, měl by uživateli dát
+možnost se připojit do daných místností, zároveň by ve svém uživatelské rozhraní
+měl zobrazit, kteří uživatelé náleží kterým místnostem \cite{GitHub-MSC3985}.
+
+Jelikož je nepraktické, aby v místnosti bylo víc událostí typu
+\mintinline{text}{m.breakout}, není to povoleno. Toho docílíme tím, že jako
+\mintinline{text}{state_key} budeme vždy využívat prázdný řetězec znaků (tedy
+\mintinline{text}{""}) \cite{GitHub-MSC3985} -- v Matrix místnosti vždy existuje
+jen jedna validní state událost s danou dvojicí \mintinline{text}{type} a
+\mintinline{text}{state_key} \cite{MatrixORG-Spec}.
+
+Na GitHubu v repositáři
+\href{https://github.com/matrix-org/matrix-spec-proposals}{matrix-spec-proposals}
+můžeme najít pull request č. 3985 --
+\href{https://github.com/matrix-org/matrix-spec-proposals/pull/3985}{MSC3985},
+kde je návrh změny specifikace veřejně dostupný \cite{GitHub-MSC3985}.
 
 \subsection{Samotná implementace}
 
-Lorem
+Pro přidání funkce break-out místností do aplikace Element Call budeme muset
+provést změny ve dvou repositářích --
+\href{https://github.com/matrix-org/matrix-js-sdk/}{matrix-js-sdk} (Matrix
+JavaScript SDK -- více v části \ref{matrix-js-sdk}) a
+\href{https://github.com/vector-im/element-call/}{element-call} (samotná
+aplikace Element Call). Jak Element Call tak Matrix JavaScript SDK využívají
+programovací jazyk TypeScript \cite{GitHub-MatrixJSSDK, GitHub-ElementCall}, což
+je nadstavba na JavaScriptu, která jej rozšiřuje o statické typování
+\cite{TypeScript-Homepage}.
+
```

Comment:
Jen dotaz: Dá nebo bude se nějak řešit to, jak donutit uživatele se do příslušných místností připojit? Např. v Zoomu lze prostě vynutit rozdělení uživatelů do break-out rooms a uživatel s tím nemůže nic dělat. A naopak po nějaké době lze klienty vynuceně vrátit zpět do hlavní místnosti. Z toho tvého popisu jakoby vyplývá, že pro klienta matrixRTC je to dobrovolné.

Reply:
Pravděpodobně nějaké řešení bude, ale bude záviset na nějakém obecném "permission modelu", který zatím není pro MatrixRTC řešený, takže jsem se tím nezabýval